File: C++/Problems/Problems/problem_2828.cpp
#include "problem_2828.h"

bool problem_2828::test() {
    std::vector<std::string> words { "alice", "bob", "charlie" };
    std::string s = "abc";

    bool expected = true;

    auto result = isAcronym(words, s);

    return expected == result;
}

bool problem_2828::isAcronym(std::vector<std::string> &words, std::string s) {
    int n = words.size();
    int m = s.size();
    if (n != m)
        return false;
    for (int i = 0; i < n; ++i)
    {
        if (words[i][0] != s[i])
            return false;
    }

    return true;
}

File: Python/Problems/Problems/problem2828.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem


class Problem2828(Problem):
    def test(self) -> bool:
        words = ["alice", "bob", "charlie"]
        s = "abc"

        expected = True

        result = self.isAcronym(words, s)

        return expected == result

    def isAcronym(self, words: List[str], s: str) -> bool:
        n = len(words)
        m = len(s)
        if n != m:
            return False
        for i in range(n):
            if words[i][0] != s[i]:
                return False

        return True